### ISSUE TITLE
Failing to generate go-client and build is not able to go past generation.

### ISSUE THREAD

**Issue by kannon92** [kubernetes-sigs/jobset#200]
With the release of the go-client and python-sdk, I am unable to build job-set on main.  

I run `make`:
```
[ec2-user@ip-172-31-93-184 jobset-kevin]$ make
test -s /home/ec2-user/Work/GIT/jobset-kevin/bin/controller-gen && /home/ec2-user/Work/GIT/jobset-kevin/bin/controller-gen --version | grep -q v0.11.4 || \
GOBIN=/home/ec2-user/Work/GIT/jobset-kevin/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.4
/home/ec2-user/Work/GIT/jobset-kevin/bin/controller-gen \
        rbac:roleName=manager-role output:rbac:artifacts:config=config/components/rbac\
        crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/components/crd/bases\
        webhook output:webhook:artifacts:config=config/components/webhook\
        paths="./..."
cp -f /home/ec2-user/go/1.20.2/pkg/mod/k8s.io/code-generator@v0.27.2/generate-groups.sh /home/ec2-user/Work/GIT/jobset-kevin/bin/
/home/ec2-user/Work/GIT/jobset-kevin/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
./hack/update-codegen.sh go /home/ec2-user/Work/GIT/jobset-kevin/bin
Generating deepcopy funcs
```

And I get a series of failures which I traced to this call:

```
[ec2-user@ip-172-31-93-184 jobset-kevin]$ ./hack/update-codegen.sh go /home/ec2-user/Work/GIT/jobset-kevin/bin
Generating deepcopy funcs
Generating apply configuration for jobset:v1alpha2 at sigs.k8s.io/jobset/client-go/applyconfiguration
F0620 21:36:53.858060 3519274 main.go:51] Error: Failed executing generator: some packages had errors:
errors in package "sigs.k8s.io/jobset/client-go/applyconfiguration/jobset/v1alpha2":
open /home/ec2-user/go/1.20.2/src/sigs.k8s.io/jobset/client-go/applyconfiguration/jobset/v1alpha2/network.go: no such file or directory
open /home/ec2-user/go/1.20.2/src/sigs.k8s.io/jobset/client-go/applyconfiguration/jobset/v1alpha2/replicatedjob.go: no such file or directory
```

Any ideas on what is wrong?

Is there a substitution that is failing somewhere that I am missing?

My bash is 5.2.0 and I am running on Amazon Linux.

**Comment by danielvegamyhre** [comment]
Hmm I'm unable to reproduce this so far - I did a fresh clone and ran `make` and it ran successfully

**Comment by danielvegamyhre** [comment]
@kannon92 are you building from the `main` branch? have you double-checked your local branch is fully up to date with the latest upstream changes?

**Comment by kannon92** [comment]
Yea.

git log:

```
commit 162e74d8155804d4acadcf831588ea5671f17c9f (HEAD -> main, upstream/main, origin/main, origin/HEAD, master)
Merge: c075cd8 d0b885e
Author: Kubernetes Prow Robot <k8s-ci-robot@users.noreply.github.com>
Date:   Wed Jun 21 17:43:41 2023 -0700

    Merge pull request #195 from kubernetes-sigs/dependabot/go_modules/github.com/onsi/ginkgo/v2-2.11.0
    
    Bump github.com/onsi/ginkgo/v2 from 2.10.0 to 2.11.0

```

I'm actually trying to get an ubuntu instance up and I'll see if I can switch over to that.  I wonder if there is some bash that is failing on my ec2 instance that isn't happening on yours or CI.

**Comment by kannon92** [comment]
> @kannon92 I just tried on a Google Cloud `n1-standard-1` instance instead of my workstation and I also didn't encounter this issue, so I suspect it something to do with your local environment. For reference, my Bash version is 5.1.4(1)-release and Go version 1.20.

:(

Not sure what is going wrong.  I tried on main and did a fresh clean.  I'll try a reclone and see.

**Comment by kannon92** [comment]
When you run it are you getting the following parameters using the codegen:

```
./hack/update-codegen.sh go /home/ec2-user/Work/GIT/jobset-kevin/bin
```

**Comment by kannon92** [comment]
> @kannon92 I just tried on a Google Cloud `n1-standard-1` instance instead of my workstation and I also didn't encounter this issue, so I suspect it something to do with your local environment. For reference, my Bash version is 5.1.4(1)-release and Go version 1.20.

my version of bash is 5.2.0.

**Comment by danielvegamyhre** [comment]
> When you run it are you getting the following parameters using the codegen:
> 
> ```
> ./hack/update-codegen.sh go /home/ec2-user/Work/GIT/jobset-kevin/bin
> ```

My output for this line is:

`./hack/update-codegen.sh go /usr/local/google/home/danielvm/go/src/sigs.k8s.io/jobset/bin`

**Comment by kannon92** [comment]
Not sure what is going on but something is off with my amazon vm for this client generation.  I have had 0 issues building anything until the client generation.  

I switched to an ubuntu instance and I was able to get everything working.  

But I noticed by default it put the generation of the client in the hack directory (/home/ubuntu/GIT/jobset/hack/sigs.k8s.io).  Not sure if that is intended.  

I am just running `make` from the top-level source directory.

**Comment by danielvegamyhre** [comment]
> A question i have is why do we have two different OS's generate the client into different locations.. Shouldn't it modify the code in client-go if there are any changes?

Yes there appears to be a bug in the output directory configuration in the bash script, it probably needs to be set relative to the repo root and currently is just relative to the current working directory or something like that.

> But I am running into some issues with the python client generation..

That looks like an issue with the python version being used, can you check what python version is being used?

**Comment by kannon92** [comment]
My python3 is 3.7.  What version did you test with?  Changing the `usr/env/python3` to use python3.10 fixed it for me.  But not sure what we are targeting.

**Comment by danielvegamyhre** [comment]
@kannon92 It should be using python 3.10, we just made a change to `test-sdk.sh` which unpins the python3 version, since it should default to 3.11 in the CI Environment, however, it appears this will break some users local development environments which default to an older python version. I think we may want to pin Python 3.10 to support the newer typing syntax we use in the python script.

**Comment by danielvegamyhre** [comment]
@kannon92 actually, if the user has some other python version (say 3.7) installed already, then test-sdk.sh installs python 3.10, we will also need to install something like pyenv for managing different python installations on the same machine... there is only 1 function signature in `post_gen.py` which uses these newer type hints (e.g. native `list` instead of `typing.List`, etc.). 

I am more inclined to simply remove the type hints on this line and keep the python installation the way it is, since removing the type hints from this 1 line in a short script is inconsequential. What do you think?

**Comment by kannon92** [comment]
Requiring 3.10 for a script seems strong to me so I think removing that would be good.
